packages/web3-eth2-core/src/index.ts

```diff
@@ -0,0 +1,89 @@
+import Axios, {AxiosInstance} from 'axios'
+
+import { ETH2BaseOpts, ETH2Function } from '../types/index'
+import { IBaseAPISchema } from './schema'
+
+export class ETH2Core {
+    private _httpClient: AxiosInstance
+
+    [ key: string ]: ETH2Function | any;
```

Comment:
Do we need `any`?

Reply:
Looking at ETH2Function, since its a generic I think we can drop `any`. Thoughts?

Reply:
This was taken from Ethers.js, I believe. Removing it causes typescript to complain that class properties such as `name`, `provider`, and `protectProvider` are not of type `ETH2Function`, which they correctly aren't, and also that some class methods such as `setProvider` aren't of type `ETH2Function` because they don't return promises. If there's a better way to do this, I'm all for it, but I believe the `any` might be a necessary evil here